src/Blacklist.php

```diff
@@ -47,17 +54,13 @@ public function __construct(Storage $storage)
     public function add(Payload $payload)
     {
         $exp = Utils::timestamp($payload['exp']);
+        $refreshExp = Utils::timestamp($payload['iat'])->addMinute($this->refreshTTL);
+        $lastExp = $exp->max($refreshExp); // the later of the two expiration dates
 
-        // there is no need to add the token to the blacklist
-        // if it has already expired
-        if ($exp->isPast()) {
-            return false;
-        }
+        // find the number of minutes until the expiration date, plus 1 minute to avoid overlap
+        $minutes = $lastExp->diffInMinutes(Utils::now()->subMinute());
 
-        // add a minute to abate potential overlap
-        $minutes = $exp->diffInMinutes(Utils::now()->subMinute());
-
-        $this->storage->add($payload['jti'], ['valid_until' => $this->getGraceTimestamp($exp)], $minutes);
+        $this->storage->add($payload['jti'], ['valid_until' => $this->getGraceTimestamp(Carbon::now())], $minutes);
```

Comment:
think it would be good to use `Utils::now()` instead of `Carbon::now()` directly just to keep things consistent